dragonchain/lib/dto/bnb.py

```diff
@@ -0,0 +1,379 @@
+        """
+        _log.info(f"[BINANCE] Getting confirmations for {transaction_hash}")
+        # Binance docs lie -- txn_hash is expected to be base64, not hex:
+        transaction_hash = base64.b64encode(bytes.fromhex(transaction_hash)).decode("ascii")
+        response = self._call_node_rpc("tx", {"hash": transaction_hash, "prove": True}).json()
+        if response.get("error") is not None:  # can't use HTTP status codes, it is a 200
+            if "not found" in response["error"]["data"]:  # txn wasn't found
+                _log.warning(f"[BINANCE] response error: {response['error']['data']}")
+                raise exceptions.TransactionNotFound(f"[BINANCE] Transaction {transaction_hash} not found")
+        transaction_block_number = int(response["result"]["height"])
+        latest_block_number = self.get_current_block()
+        _log.info(f"[BINANCE] Latest block number: {latest_block_number} | Block number of transaction: {transaction_block_number}")
+        return (latest_block_number - transaction_block_number) >= CONFIRMATIONS_CONSIDERED_FINAL
+
+    # https://docs.binance.org/api-reference/api-server.html#apiv1balanceaddresssymbol
+    def check_balance(self, symbol: str = "BNB") -> int:
+        """Check the balance of the address for this network
+        Args:
+            symbol: stock ticker symbol for token you're trying to check the balance of
+        Returns:
+            The amount of funds of that token in the wallet
+        """
+        _log.info(f"[BINANCE] Checking {symbol} balance for {self.address}")
+        path = f"balances/{self.address}/{symbol}"  # params expected inside path string
+        response = self._call_node_api(path)
+        response_json = response.json()
+        if response_json.get("error") is not None:  # can't use HTTP status codes, it is a 200
+            if "interface is nil, not types.NamedAccount" in response_json["error"]["data"] and response.status_code == 500:
+                _log.warning(f"[BINANCE] Non 200 response from Binance node:")
+                _log.warning(f"[BINANCE] response code: {response.status_code}")
+                _log.warning(f"[BINANCE] response error: {response_json['error']['data']}")
+                _log.warning("[BINANCE] This is actually expected for a zero balance address.")
+                return 0  # return a zero balance
+        bnb_balance = int(response_json["balance"]["free"])
+        return bnb_balance
+
+    # https://docs.binance.org/api-reference/api-server.html#apiv1fees
+    def get_transaction_fee_estimate(self) -> int:
+        """Calculate the transaction fee estimate for a transaction given current fee rates
+        Returns:
+            The amount of estimated transaction fee cost for the network
+        """
+        _log.info("[BINANCE] Fetching the send fee...")
+        response = self._call_node_api("fees")  # don't .json(); "fees" endpoint returns a list!
```

Comment:
this definitely doesn't work

Reply:
`fees` endpoint returns a list, but this can still be parsed from the Response object as JSON.  Tested with Python REPL.  Fixed.